### DISCUSSION TITLE
actix_web::main vs tokio::main

### DISCUSSION THREAD

**Discussion by pinwheel101** [actix/actix-web#2852] [Q&A] [answered]
I am very new to Rust and Actix.

What is the difference to use between `#[actix_web::main]` vs `#[tokio::main]`?

Seems like it choose the runtime between `actix-rt` and `tokio`.

Based on crates.io explanation, actix-rt is single threaded async runtime and tokio is multi-thread.

What is pros and cons? (It is very difficult to figure out for me)

(Is there any link(s) to study this topic?)

**Comment by robjtede** [comment, marked as answer]
This page should give some good info: https://docs.rs/actix-web/latest/actix_web/rt/index.html#running-actix-web-using-tokiomain

Lmk if there's something missing that we can add to the page.

**Reply by pinwheel101** [reply]
@robjtede  Thank you so much ;). The link is very helpful.
